lib/web_ui/lib/src/engine/js_interop/js_loader.dart

```diff
@@ -75,7 +75,7 @@ abstract class FlutterEngineInitializer{
 @staticInterop
 abstract class FlutterAppRunner {
   factory FlutterAppRunner({required RunAppFn runApp,}) => FlutterAppRunner._(
-    runApp: (([RunAppFnParameters? args]) => futureToPromise(runApp(args) as Future<JSObject>)).toJS
+    runApp: (([RunAppFnParameters? args]) => (runApp(args) as Future<JSObject>).toPromise).toJS
```

Comment:
If I remove the `as` cast, `toPromise` stops working (?)

Reply:
> `FlutterApp` isn't a JS type since it isn't an interop extension type

This makes sense to me. Next steps would be to start migrating this whole js_interop directory to the proper extension-type based interop types.

(That's another Yak, though :P)